rust_src/src/buffers.rs

```diff
@@ -320,6 +335,57 @@ pub fn overlay_buffer(overlay: LispObject) -> LispObject {
     marker_buffer(marker)
 }
 
+pub fn recenter_overlay_lists(mut buf: LispBufferRef, pos: ptrdiff_t) {
+    // let mut overlay: LispObject;
+    let (mut prev, mut tail, mut next): (
+        Option<LispOverlayRef>,
+        LispOverlayRef,
+        Option<LispOverlayRef>,
+    ) = (
+        None,
+        LispOverlayRef::new((*buf).overlays_before as *mut Lisp_Overlay),
+        None,
+    );
+    loop {
+        if tail.as_ptr() as ptrdiff_t == 0 {
```

Comment:
Is this a check for `ptr::null()'?

Reply:
Actually, this is the check for the condition in the [`for` loop](https://github.com/Wilfred/remacs/blob/aabf185e251f8586e1e4003af7fc388dcbf537e8/src/buffer.c#L3338), which indeed seems to be a `ptr::null()` check. How would you write this?

Reply:
`tail.is_null()`

Reply:
Instead of a `loop` why not use `while !tail.is_null()`? Loop means we have to really check for run away infinite loops.